### PR TITLE
Merge ExtendedQueryTag feature branch back to main

### PR DESCRIPTION
**Pull request by pengchen0692**
## Description
Merge ExtendedQueryTag feature branch back to main

## Related issues
Addresses [AB#80255](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/80255)

## Testing
All automation tests pass

## Manual Merge Files
test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/HttpIntegrationTestFixture.cs
test/Microsoft.Health.Dicom.Web.Tests.E2E/Microsoft.Health.Dicom.Web.Tests.E2E.csproj
test/Microsoft.Health.Dicom.Web.Tests.E2E/Common/InProcTestDicomWebServer.cs
test/Microsoft.Health.Dicom.Tests.Integration/Persistence/SqlDataStoreTestsFixture.cs
test/Microsoft.Health.Dicom.Tests.Integration/Microsoft.Health.Dicom.Tests.Integration.csproj
src/Microsoft.Health.Dicom.SqlServer/Features/Query/SqlQueryGenerator.cs
src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj
src/Microsoft.Health.Dicom.Metadata/Microsoft.Health.Dicom.Metadata.csproj
src/Microsoft.Health.Dicom.Core/Microsoft.Health.Dicom.Core.csproj
src/Microsoft.Health.Dicom.Core/Features/Store/StoreOrchestrator.cs
src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/ExtendedQueryTagEntryValidator.cs
src/Microsoft.Health.Dicom.Core/Features/Common/DicomTagParser.cs
src/Microsoft.Health.Dicom.Core/DicomCoreResource.resx
src/Microsoft.Health.Dicom.Core/DicomCoreResource.Designer.cs
src/Microsoft.Health.Dicom.Core.UnitTests/Microsoft.Health.Dicom.Core.UnitTests.csproj
src/Microsoft.Health.Dicom.Client/DicomWebClient.cs
src/Microsoft.Health.Dicom.Blob/Microsoft.Health.Dicom.Blob.csproj
src/Microsoft.Health.Dicom.Api/Microsoft.Health.Dicom.Api.csproj
converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Microsoft.Health.DicomCast.Core.UnitTests.csproj
Directory.Build.props
.editorconfig

### CODE REVIEW

File: test/Microsoft.Health.Dicom.Tests.Integration/Microsoft.Health.Dicom.Tests.Integration.csproj [hunk 1]
```diff
@@ -27,6 +27,7 @@
     <PackageReference Include="Polly" Version="7.2.1" />
     <PackageReference Include="xunit" Version="2.4.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
+    <PackageReference Include="Microsoft.SqlServer.DACFx" Version="150.4897.1" />
```
Comment (wsugarman) [inline test/Microsoft.Health.Dicom.Tests.Integration/Microsoft.Health.Dicom.Tests.Integration.csproj:30]: Is this required?

File: src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj [hunk 2]
```diff
@@ -14,8 +14,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\Microsoft.Health.Dicom.Core\Microsoft.Health.Dicom.Core.csproj" />
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj:17]: Is Core no longer necessary?

File: src/Microsoft.Health.Dicom.Core.UnitTests/Microsoft.Health.Dicom.Core.UnitTests.csproj [hunk 3]
```diff
@@ -24,7 +24,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\test\Microsoft.Health.Dicom.Tests.Integration\Microsoft.Health.Dicom.Tests.Integration.csproj" />
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core.UnitTests/Microsoft.Health.Dicom.Core.UnitTests.csproj:27]: Is this no longer a dependency?

File: src/Microsoft.Health.Dicom.Core/DicomCoreResource.Designer.cs [hunk 4]
```diff
@@ -87,15 +87,6 @@ internal class DicomCoreResource {
             }
         }
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core/DicomCoreResource.Designer.cs:89]: Just double-check that this file has been successfully regenerated

**Review by wsugarman — APPROVED** [review]

File: test/Microsoft.Health.Dicom.Tests.Integration/Microsoft.Health.Dicom.Tests.Integration.csproj (hunk 1 above)
Comment (pengchen0692) [reply test/Microsoft.Health.Dicom.Tests.Integration/Microsoft.Health.Dicom.Tests.Integration.csproj:30]: yes, it's required for database schema comparison https://github.com/microsoft/dicom-server/blob/feature/customtags/test/Microsoft.Health.Dicom.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs

Comment (pengchen0692) [reply test/Microsoft.Health.Dicom.Tests.Integration/Microsoft.Health.Dicom.Tests.Integration.csproj:30]: it's copied from FHIR https://github.com/microsoft/fhir-server/blob/8fb192f6a9fc4fad735182e683340e6b5900e92a/test/Microsoft.Health.Fhir.R5.Tests.Integration/Microsoft.Health.Fhir.R5.Tests.Integration.csproj#L10

File: src/Microsoft.Health.Dicom.Core.UnitTests/Microsoft.Health.Dicom.Core.UnitTests.csproj (hunk 3 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Core.UnitTests/Microsoft.Health.Dicom.Core.UnitTests.csproj:27]: yeah, Unit test library should not have depends on integration test.

File: src/Microsoft.Health.Dicom.Core/DicomCoreResource.Designer.cs (hunk 4 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Core/DicomCoreResource.Designer.cs:89]: sure

File: src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj (hunk 2 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj:17]: added back

**Pull request merged**